--- jdk16u-jdk-16.0.0-0/src/hotspot/os/solaris/os_solaris.cpp
+++ jdk16u-jdk-16.0.0-0/src/hotspot/os/solaris/os_solaris.cpp
@@ -168,12 +168,6 @@
 
 os::Solaris::pthread_setname_np_func_t os::Solaris::_pthread_setname_np = NULL;
 
-// "default" initializers for missing libc APIs
-extern "C" {
-  int memcntl(caddr_t, size_t, int, caddr_t, int, int);
-  int meminfo(const uint64_t *, int, const uint_t *, int, uint64_t *, uint_t *);
-}
-
 // "default" initializers for pthread-based synchronization
 extern "C" {
   static int pthread_mutex_default_init(mutex_t *mx, int scope, void *arg) { memset(mx, 0, sizeof(mutex_t)); return 0; }
@@ -2242,7 +2236,7 @@
   mpss_struct.mha_pagesize = align;
   mpss_struct.mha_flags = 0;
   // Upon successful completion, memcntl() returns 0
-  if (memcntl(start, bytes, MC_HAT_ADVISE, (caddr_t) &mpss_struct, 0, 0)) {
+  if (memcntl(start, bytes, MC_HAT_ADVISE, (void*) &mpss_struct, 0, 0)) {
     debug_only(warning("Attempt to use MPSS failed."));
     return false;
   }
