--- jdk15u-jdk-15.0.1-12/src/hotspot/os/solaris/os_solaris.cpp
+++ jdk15u-jdk-15.0.1-12/src/hotspot/os/solaris/os_solaris.cpp
@@ -169,8 +169,6 @@
 
   static int lwp_cond_init(cond_t *cv, int scope, void *arg){ memset(cv, 0, sizeof(cond_t)); return 0; }
   static int lwp_cond_destroy(cond_t *cv)                   { return 0; }
-  int memcntl(caddr_t, size_t, int, caddr_t, int, int);
-  int meminfo(const uint64_t *, int, const uint_t *, int, uint64_t *, uint_t *);
 }
 
 // "default" initializers for pthread-based synchronization
@@ -2589,7 +2587,7 @@
   mpss_struct.mha_pagesize = align;
   mpss_struct.mha_flags = 0;
   // Upon successful completion, memcntl() returns 0
-  if (memcntl(start, bytes, MC_HAT_ADVISE, (caddr_t) &mpss_struct, 0, 0)) {
+  if (memcntl(start, bytes, MC_HAT_ADVISE, (void*) &mpss_struct, 0, 0)) {
     debug_only(warning("Attempt to use MPSS failed."));
     return false;
   }
